### PR TITLE
Refactor labeler code to allow usage without Azure dependencies (+ other small fixes)

### PR DESCRIPTION
**Pull request by Eilon**
Ideally, look at each commit individually. Most are just a few lines. The big one is the refactoring that has this useful comment:

> Refactor Predictor code so it can be used without Azure dependencies
> 1. New LabelPredictor project has all the ML logic and GitHub awareness, but no dependencies on Azure queues, blobs, or KeyVault.
> 2. New interfaces in LabelPredictor project, along with default "local file" based providers
> 3. Original IssueLabeler Web API project implements the interfaces to be Azure-enabled.
> 
> By default there should be no behavior changes at all - everything should work exactly as it did before. But you can swap the commented code in Startup.cs to change the behavior for local testing, or use the LabelPredictor project in other scenarios that don't use Azure storage.

The refactoring is 95% just moving code and renaming some types. The few small _changes_ were to add the new interfaces and localize the Azure dependencies into Azure-aware types. Then there was a bit of new code for the "local file" implementations of those interfaces (which is quite trivial).

### CODE REVIEW

File: src/LabelPredictor/IGitHubClientFactory.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Octokit;
```
Comment (maryamariyan) [inline src/LabelPredictor/IGitHubClientFactory.cs:1]: We'd need the header license on the new files:

```cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
```

**Review by maryamariyan — APPROVED** [review]

File: src/LabelPredictor/GitHubClientWrapper.cs [hunk 2]
```diff
@@ -19,18 +18,14 @@ public class GitHubClientWrapper : IGitHubClientWrapper
     {
         private readonly ILogger<GitHubClientWrapper> _logger;
         private GitHubClient _client;
-        private readonly GitHubClientFactory _gitHubClientFactory;
-        private readonly bool _skipAzureKeyVault;
```
[the remark this answers is not in the document]
Comment (maryamariyan) [reply src/LabelPredictor/GitHubClientWrapper.cs:23]: Marking resolved, seems like this is moved to `AzureKeyVaultGitHubClientFactory`

**Comment by Eilon** [comment]
License headers updated. When you're ready to merge, please use **rebase** instead of **squash** because some of the individual commits in this PR are useful as-is and I'd rather not merge them. Thank you!

**Comment by Eilon** [comment]
image

**Pull request merged**